### PR TITLE
Desktop: Rework idle behavior, fix popups, adjust lock and login screen styling

### PR DESCRIPTION
**Pull request by kajusnau**
## Description of Changes

1. **Reworked System Idle Behavior**  
   - Implemented a multi-stage idle workflow:  
     - **Dim:** Notification is shown on the focused display and screen dims after 80% of `autolock.duration` (4 minutes, with default `autolock.duration` of 300 seconds).  
     - **Lock:** Locks session after 100% of duration (5 minutes).  
     - **Screen Off:** Turns off the screen after 150% of duration (7 minutes 30 seconds).  
     - **Suspend:** Suspends the system after 300% of duration (15 minutes).  
   - Each stage can be interrupted by user actions, such as moving the mouse or pressing a key, to reset the idle timer.

2. **Lock Screen and Login Screen Enhancements**  
   - Adjusted the styling of the lock and login screens.
   - Added gtklock powerbar module with the following buttons enabled - poweroff, reboot, suspend, logout.  
   - Added a custom layout for the lock screen.  

3. **Configuration Improvements**  
   - Added a separate configuration file for `gtklock`.
   - Added a separate configuration file for `swayidle`.  
   - Enabled titlebar icons in `labwc` configuration.

4. **Improved Volume and Brightness Popups**  
   - Popups now only appear when triggered by function key presses.  
   - Prevents unnecessary popups, ensuring they only show when expected by the user.

5. **Taskbar and Power Menu Adjustments**  
   - Rearranged buttons in the taskbar power menu into a more logical order. (Thanks Milla)  
   - Hidden the taskbar keyboard layout indicator while backend support is unavailable.

6. **Package and Script Updates**  
   - Replaced the unused `powercontrol` package with the `ghaf-powercontrol` GIVC wrapper script.  
   - Use the new `ghaf-powercontrol` script wherever applicable.  
   - Removed the `nm-launcher` package, preferring to launch `nm-applet` directly using a DBUS proxy.

7. **Removed Unused Waybar** 

## Checklist for things done

- [x] Summary of the proposed changes in the PR description
- [x] More detailed description in the commit message(s)
- [x] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [x] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [ ] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `make-checks` and it passes
- [x] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [x] Author has added reviewers and removed PR draft status
- [ ] Change requires full re-installation
- [x] Change can be updated with `nixos-rebuild ... switch`

## Instructions for Testing

- [ ] List all targets that this applies to: 
- [x] Is this a new feature  
  - Partially.  
    - **New Features:**  
      - Multi-stage idle workflow.  
      - Separate config files for `swayidle` and `gtklock`.  
      - Custom lock screen layout.  
    - **Improvements:**  
      - Enhanced volume/brightness popups, taskbar power menu order, and system config updates.  
  - [ ] List the test steps to verify:  
    1. **Idle Behavior Testing:**  
       - Ensure the following behaviors occur, based on the default `autolock.duration` value of 300s:  
         - Notification is shown on the focused display and screen dims after 80% of the set duration.  
         - Locks session after 100% of the set duration.  
         - Turns off the screen after 150% of the set duration.  
         - Suspends the system after 300% of the set duration.  
       - Confirm user interaction (e.g., mouse movement) interrupts the process at each stage.  
    2. **Lock Screen Enhancements:**  
       - Confirm the new look and feel of the lock screen based on the PR description.
       - Confirm powerbar module is visible and functions as expected (should appear near the bottom of the lock screen).
    3. **Volume/Brightness Popups:**  
       - Test triggering popups using function keys (F1, F2, F3, F5, F6).  
       - Confirm no unexpected popups appear during normal system usage, especially when opening MS Teams, Chrome, etc.  
    4. **Taskbar Power Menu and Layout Indicator:**  
       - Ensure power menu buttons are in the updated logical order.  
       - Confirm the keyboard layout indicator is hidden.  
    5. **Script and Package Updates:**  
       - Verify no regression in using `nm-applet` from the system tray.  
       - Verify no regression in system power control - taskbar and lock screen power state buttons should function as expected.  
- [x] If it is an improvement how does it impact existing functionality?  
  - Simplifies and organizes idle behavior for a more intuitive user experience.  
  - Improves lock screen and login screen aesthetics and functionality.

### CODE REVIEW

**Review by brianmcgillion — APPROVED** [review]
Approve with prejudice :)

**Comment by milva-unikie** [comment]
I have with Kajus already done a couple of rounds of testing for this PR with Lenovo-X1. There is one more small popup issue that still needs to be fixed.

Overall I really like the improvements this PR brings to Ghaf!

File: modules/desktop/graphics/waybar.config.nix [hunk 1]
```diff
@@ -87,7 +87,7 @@ in
             "interval": 15,
             "exec": "${wifi-signal-strength}/bin/wifi-signal-strength",
```
[the remark this answers is not in the document]
Comment (kajusnau) [reply modules/desktop/graphics/waybar.config.nix:88]: Updated just in case as I wasn't sure.
I'll remove it here with next update

**Comment by johannarautanen** [comment]
**Tested with native Orin AGX and NX.** Flashed to memory card

Working:
- Taskbar Power Menu; power menu buttons are in the updated logical order: lock, suspend, logout, reboot, shudown. Buttons worked ok.
- In lock-screen were are following buttons enabled - poweroff, reboot, suspend, logout.  And there worked as they should.
- When lock is activated, there appeared first time and after screen activation the lock screen come visible.
- "win + L" locked the screen
- there is "caps lock is on" visible in lock screen when caps lock activated and "login failed" comes after entering wrong lock code.
- there are no any extra staff visible in Orin after this change

- automated test worked.
- applications can be launched

**Comment by johannarautanen** [comment]
Tested with crosscompile Orin AGX and NX. Flashed to memory card

Working:

- Taskbar Power Menu; power menu buttons are in the updated logical order: lock, suspend, logout, reboot, shudown. Buttons worked ok.
- In lock-screen were are following buttons enabled - poweroff, reboot, suspend, logout. And there worked as they should.
- When lock is activated, there appeared first time and after screen activation the lock screen come visible.
- "win + L" locked the screen
- there is "caps lock is on"-text visible in lock screen when caps lock activated and "login failed" comes after entering wrong lock code.
- there are no any extra staff visible in Orin after this change
- applications can be launched

**Comment by milva-unikie** [comment]
Tested on Lenovo-X1 (`nixos-rebuild switch`)
- Everything mentioned in testing instructions work
- Test-automation passes
- Looks great
- Works well
- No complaints :)

**Pull request merged**